### PR TITLE
Add API reference for Astro.cookies

### PR DESCRIPTION
**Pull request by matthewp**
#### What kind of changes does this PR include?

- New or updated content

Docs for `Astro.cookies`

#### Description

Docs for https://github.com/withastro/astro/pull/4876

### CODE REVIEW

**Review by sarah11918 — COMMENTED** [review]
Yay, thanks for getting this together so promptly, @matthewp! Just some thoughts/suggestions below. This content isn't really my area, so will also wait for @Jutanium / @delucis to weigh in with more notable commentary.

File: src/pages/en/reference/api-reference.md
Comment (matthewp) [reply src/pages/en/reference/api-reference.md:177]: Am I reading this right, or is the suggestion to remove the `set` documentation?

For options we use this library and just pass through our options to it: https://www.npmjs.com/package/cookie#options-1

Do we prefer linking in cases like this or to repeat the information?

Comment (matthewp) [reply src/pages/en/reference/api-reference.md:177]: I think it depends on where/when you use them. It's common to use `get` on pages, to do things like check if a user is logged in to determine if you should redirect them to a login page.

`set` is most of the time only going to be done on non-`GET` methods. So more likely you'll do those in API Routes.

Comment (sarah11918) [reply src/pages/en/reference/api-reference.md:177]: For the options library, I'd link the words `cookie features` to npmjs.com

File: src/pages/en/reference/api-reference.md
Comment (matthewp) [reply src/pages/en/reference/api-reference.md:178]: So technically what is happening is when you "delete" a cookie you are really just sending a `Set-Cookie: key=value; expires=Thu, 01 Jan 1970 00:00:00 GMT`. Note here that `expires` is in the past, this tells the browser that the cookie has expired so it deletes it.

So you're really just telling the browser that the cookie is no longer valid.

I'm not sure if we need to explain that part to the user here? I will defer to your judgement on that.

What I was trying to convey here is that, from the context of the `Astro.cookies` API, once you call `delete(key)` that cookie no longer exists, you cannot do `get(key)` or `has(key)` to get its value. It's "deleted" from memory, if you will.

File: src/pages/en/reference/api-reference.md
Comment (sarah11918) [reply src/pages/en/reference/api-reference.md]: Fair enough! Fine by me!

Comment (matthewp) [reply src/pages/en/reference/api-reference.md]: Added this 👍

File: src/pages/en/reference/api-reference.md
Comment (sarah11918) [reply src/pages/en/reference/api-reference.md:178]: So, what about something like:

> Marks the cookie **as** deleted. Once a cookie is deleted, `Astro.cookies.has()` will return false and `Astro.cookies.get() will .... .  (throw an error/return null.. whatever it does)

Would that describe it well enough? (Using "as" to imply that it "takes effect" immediately? Could also say "as expired" or "as invalid") Then, describe what the other functions will return?

**Review by delucis — APPROVED** [review]
LGTM! Thanks, Matthew.

**Comment by sarah11918** [comment]
Just reminding you @matthewp that this is your PR to merge when you're happy with it! (In Docs, we always leave it up to the author to merge, if they have maintainer authority.)

File: src/pages/en/reference/api-reference.md
Comment (delucis) [inline src/pages/en/reference/api-reference.md:170]: Ah, we should probably include this to flag the version this API was added in. Could help anyone on an older version who's looking at these docs.

```suggestion

<Since v="1.4.0" />

```

File: src/pages/en/reference/api-reference.md
Comment (matthewp) [reply src/pages/en/reference/api-reference.md]: Will keep it as is for now, I don't care if people change the order in the future though, no strong opinions on it from me.

File: src/pages/en/reference/api-reference.md
Comment (matthewp) [reply src/pages/en/reference/api-reference.md:180]: Yeah I like this, it skips over the "how" which is not important from the API perspective and just focuses on how Astro.cookies will work after deletion. I'll update with something like this.

**Pull request merged**
